src/components/UserSelector.tsx

```diff
@@ -1,19 +1,106 @@
-import React from 'react';
+import React, {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { getUsers } from '../utils/serverHelper';
+
+type Props = {
+  setUserSelectedId: (id: number) => void,
+  userSelectedId: number,
+  setIsVisiblePostDetails: (value: boolean) => void,
+  setIsVisibleSideBar: (value: boolean) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  setUserSelectedId,
+  userSelectedId,
+  setIsVisiblePostDetails,
+  setIsVisibleSideBar,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [showUsers, setShowUsers] = useState(false);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const loadUsers = async () => {
```

Comment:
useCallbeck ?

Reply:
@GarryRocksman to be honest, i don't understand where it's better to use in that case?

Reply:
@Malva37 for now this function will be creating on each component rendering, and you need it only once (on component did mount action), so you can wrap it via `useCallback` to avoid it's recreation on each rendering

in this component such recreation won't affect on anything, just on memory usage

Reply:
Thank for explanation)